### PR TITLE
Use total_found instead of total

### PR DESCRIPTION
**Pull request by lmuzinic**
Could this be a better fit? 

According to http://sphinxsearch.com/docs/current.html
"total" is total amount of matches retrieved on server, while
"total_found" is total amount of matching documents in index

### CODE REVIEW

**Comment by klimov-paul** [comment]
> Could this be a better fit?

Do you have a proof it is better fit?

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,6 @@ phpunit.phar
 
 # local tests configuration
 /tests/data/config.local.php
+
+# runtime cache
+/tests/runtime/cache
```
Comment (cebe) [inline .gitignore]: should be just `/tests/runtime`.

**Comment by klimov-paul** [comment]
@lmuzinic, can you confirm 'total_found' returns same result as `Query::count()` for large amount of records, which exceeds 'max_matches' restriction?
Is  'total_found' value is reliable in case 'MATCH' operator involved in the query?

**Comment by lmuzinic** [comment]
Sure, let me try reproducing it with tests on a larger index...

File: .gitignore (hunk 1 above)
Comment (lmuzinic) [inline .gitignore]: changed, tnx!

**Comment by lmuzinic** [comment]
@klimov-paul take a look, I had to change a lot of test cases as there are more entries now.

Also, Sphinx is dropping max_matches in config files, leaving it as OPTION param.

File: tests/ExternalActiveRelationTest.php [hunk 2]
```diff
@@ -49,14 +57,14 @@ public function testFindEager()
     {
         // has one :
         $articles = ArticleIndex::find()->with('source')->all();
-        $this->assertEquals(2, count($articles));
+        $this->assertEquals($this->pagination->defaultPageSize, count($articles));
```
Comment (klimov-paul) [inline tests/ExternalActiveRelationTest.php]: This is not a proper assertion.
Exact number match should be used as it was before.

File: tests/ActiveRelationTest.php [hunk 3]
```diff
@@ -49,7 +57,7 @@ public function testFindEager()
     public function testFindCompositeLink()
     {
         $articles = ArticleIndex::find()->with('sourceCompositeLink')->all();
-        $this->assertEquals(2, count($articles));
+        $this->assertEquals($this->pagination->defaultPageSize, count($articles));
```
Comment (klimov-paul) [inline tests/ActiveRelationTest.php]: This is not a proper assertion.
Exact number match should be used as it was before.

File: tests/ExternalActiveRelationTest.php [hunk 4]
```diff
@@ -13,11 +14,18 @@
  */
 class ExternalActiveRelationTest extends TestCase
 {
+    /**
+     * @var Pagination
+     */
+    protected $pagination;
```
Comment (klimov-paul) [inline tests/ExternalActiveRelationTest.php]: Should be removed

File: tests/ActiveRelationTest.php [hunk 5]
```diff
@@ -12,11 +13,18 @@
  */
 class ActiveRelationTest extends TestCase
 {
+    /**
+     * @var Pagination
+     */
+    protected $pagination;
```
Comment (klimov-paul) [inline tests/ActiveRelationTest.php]: Should be removed

File: tests/ActiveRecordTest.php [hunk 6]
```diff
@@ -12,10 +13,17 @@
  */
 class ActiveRecordTest extends TestCase
 {
+    /**
+     * @var Pagination
+     */
+    protected $pagination;
```
Comment (klimov-paul) [inline tests/ActiveRecordTest.php]: Should be removed

File: tests/ActiveRecordTest.php [hunk 7]
```diff
@@ -36,7 +44,7 @@ public function testFind()
 
         // find all
         $articles = ArticleIndex::find()->all();
-        $this->assertEquals(2, count($articles));
+        $this->assertEquals($this->pagination->defaultPageSize, count($articles));
```
Comment (klimov-paul) [inline tests/ActiveRecordTest.php]: This is not a proper assertion.
Exact number match should be used as it was before.

File: tests/ActiveRecordTest.php [hunk 8]
```diff
@@ -92,17 +101,17 @@ public function testFind()
 
         // indexBy
         $articles = ArticleIndex::find()->indexBy('author_id')->orderBy('id DESC')->all();
-        $this->assertEquals(2, count($articles));
-        $this->assertTrue($articles['1'] instanceof ArticleIndex);
-        $this->assertTrue($articles['2'] instanceof ArticleIndex);
+        $this->assertEquals($this->pagination->defaultPageSize, count($articles));
```
Comment (klimov-paul) [inline tests/ActiveRecordTest.php]: Same here

File: tests/ActiveRecordTest.php [hunk 9]
```diff
@@ -92,17 +101,17 @@ public function testFind()
...
+        $this->assertTrue($articles['1002'] instanceof ArticleIndex);
 
         // indexBy callable
         $articles = ArticleIndex::find()->indexBy(function ($article) {
             return $article->id . '-' . $article->author_id;
         })->orderBy('id DESC')->all();
-        $this->assertEquals(2, count($articles));
-        $this->assertTrue($articles['1-1'] instanceof ArticleIndex);
-        $this->assertTrue($articles['2-2'] instanceof ArticleIndex);
+        $this->assertEquals($this->pagination->defaultPageSize, count($articles));
```
Comment (klimov-paul) [inline tests/ActiveRecordTest.php]: And here

File: tests/ActiveRecordTest.php [hunk 10]
```diff
@@ -110,11 +119,11 @@ public function testFindBySql()
         // find one
         $article = ArticleIndex::findBySql('SELECT * FROM yii2_test_article_index ORDER BY id DESC')->one();
         $this->assertTrue($article instanceof ArticleIndex);
-        $this->assertEquals(2, $article->author_id);
+        $this->assertEquals(1002, $article->author_id);
 
         // find all
         $articles = ArticleIndex::findBySql('SELECT * FROM yii2_test_article_index')->all();
-        $this->assertEquals(2, count($articles));
+        $this->assertEquals($this->pagination->defaultPageSize, count($articles));
```
Comment (klimov-paul) [inline tests/ActiveRecordTest.php]: And here

**Comment by klimov-paul** [comment]
Also, CHANGELOG entry should be added.

**Comment by klimov-paul** [comment]
I just realized you have not created a separated branch.
@lmuzinic, please rework your pull request from scratch following <a href="https://github.com/yiisoft/yii2/blob/master/docs/internals/git-workflow.md">Workflow for Yii contributors</a>

**Pull request closed without merging**